api/include/opentelemetry/baggage/propagation/baggage_propagator.h

```diff
@@ -21,7 +21,11 @@ class BaggagePropagator : public opentelemetry::context::propagation::TextMapPro
               const opentelemetry::context::Context &context) noexcept override
   {
     auto baggage = opentelemetry::baggage::GetBaggage(context);
-    carrier.Set(kBaggageHeader, baggage->ToHeader());
+    auto header  = baggage->ToHeader();
+    if (header.size())
```

Comment:
Do we need do something similar in line 36 (new file), and ignore call to `SetBaggage` if `FromHeader()` returned an empty or invalid baggage?

Reply:
Yes, was thinking so, but it would need more changes in Context API - to add a copy constructor. Just tried to API changes as of now.

Reply:
Ok seems the default copy constructor is usable for context::Context, have done the changes as requested.